exporter/doc/show.go

```diff
@@ -0,0 +1,164 @@
+package doc
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+	"time"
+
+	"github.com/projectcypress/cdatools/models"
+)
+
+type Authenticator struct {
+	models.Authenticator
+	Author
+}
+
+func NewAuthenticator(a models.Authenticator) Authenticator {
+	return Authenticator{
+		Authenticator: a,
+		Author:        NewAuthor(a.Author),
+	}
+}
+
+type Header struct {
+	models.Header
+	Authenticator
+	Authors
+}
+
+func NewHeader(h models.Header) Header {
+	return Header{
+		Header:        h,
+		Authenticator: NewAuthenticator(h.Authenticator),
+		Authors:       NewAuthors(h.Authors),
+	}
+}
+
+type Record struct {
+	models.Record
+	ProviderPerformances
+}
+
+func NewRecord(r models.Record) Record {
+	return Record{
+		Record:               r,
+		ProviderPerformances: NewProviderPerformances(r.ProviderPerformances),
+	}
+}
+
+type Cat3Data struct {
+	Header
+	Record
+	Measures       models.Measure
+	MeasureSection MeasureSection
+	StartDate      int64
+	EndDate        int64
+	Timestamp      int64
+}
+
+func NewCat3Data(h models.Header, ms MeasureSection, m models.Measure, start int64, end int64) Cat3Data {
+	timeNow := time.Now().UTC().Unix()
+	return Cat3Data{
+		Header:         NewHeader(h),
+		Measures:       m,
+		MeasureSection: ms,
+		StartDate:      start,
+		EndDate:        end,
+		Timestamp:      timeNow,
+	}
+}
+
+func (c Cat3Data) Print() string {
```

Comment:
@mattrbianchi And I were talking about it, and we definitely wanted to do it at some point. Weren't sure if we wanted to do it for this PR, but basically the idea would be to "extend" all of these template structs with a single type, and this print function would be a method off that type.